runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -4001,23 +4001,51 @@ class INTERPRETER_CLASS
 		return EXECUTE_BYTECODE;
 	}
 
-	/* ToDo: unimplemented - https://github.com/eclipse-openj9/openj9/issues/13614 */
 	/* jdk.internal.misc.Unsafe: public native <V> V uninitializedDefaultValue(Class<?> clz); */
 	VMINLINE VM_BytecodeAction
 	inlUnsafeUninitializedDefaultValue(REGISTER_ARGS_LIST)
 	{
+		j9object_t cls = *(j9object_t*)_sp;
+
 		updateVMStruct(REGISTER_ARGS);
-		Assert_VM_unreachable();
+		buildInternalNativeStackFrame(REGISTER_ARGS);
+
+		j9object_t result = NULL;
+
+		/* ToDo: update this function to have the same behavior as OpenJDK when cls is null or not a vlauetype (currently OpenJDK segfaults in both those scenarios) */
+		if (NULL != cls) {
+			J9Class *j9clazz = J9VM_J9CLASS_FROM_HEAPCLASS(_currentThread, cls);
+			if (J9_IS_J9CLASS_VALUETYPE(j9clazz)) {
+				result = j9clazz->flattenedClassCache->defaultValue;
```

Comment:
Did you try a simple test to see if `flattenedClassCache->defaultValue` has already been set when using it here ? 
It is set here:
https://github.com/eclipse-openj9/openj9/blob/13ca3998af9299612a0d010e6dee59f33cc48e86/runtime/vm/ClassInitialization.cpp#L527-L528

Reply:
I asked Ehren to check the behavior of OpenJDK to see if null can be returned by this API.

Reply:
Another thing to check is to see if this API triggers verification and initialization of the value type class in OpenJDK. You can run with option `-Xlog:class+init=info` and `-Xlog:verification=info` to find out.

Reply:
OpenJDK appears to always return a valid default object even if the class has never been used before. For example, the following code will print "0" when running with OpenJDK. 

However, when using my implementation of uninitializedDefaultValue this code will throw a NullPointerException

```
import jdk.internal.misc.Unsafe;
import java.lang.reflect.Field;



primitive class Point {
	public static void main(String[] args) throws Throwable {
		Unsafe unsafe = Unsafe.getUnsafe();

		OtherPoint asdf = unsafe.uninitializedDefaultValue(OtherPoint.class);
		System.out.println(asdf.x.i);
	}
}


primitive class ValueTypeInt {
	ValueTypeInt(int i) {this.i = i;}
	final int i;
}

primitive class OtherPoint {
	final ValueTypeInt x, y;

	OtherPoint(ValueTypeInt x, ValueTypeInt y) { 
		this.x = x; 
		this.y = y; 
	}
}
```

Reply:
It also looks like OpenJDK is not doing any verification or initialization. I ran the same code I sent above with `-Xlog:class+init=info` and `-Xlog:verification=info` and the output did not mention `OtherPoint` at all

Reply:
@hangshao0 pointed out that my version of OpenJDK was out of date. After updating I am now also seeing a NullPointerException even when running the above code with OpenJDK. So my implementation of uninitializedDefaultValue is actually consistent with OpenJDK.

Reply:
> NullPointerException even when running the above code with OpenJDK.

Where is the NPE thrown on `asdf.x` or `x.i`? Can you show the stack trace

Can you try this example with a larger flattened type. Ie.

```
primitive class OtherPoint { final ReallyLargeVT x, y; ... //or use an option to disable flattening for the type
```

This is to determine if in cases where the non-nullable (AKA flattened type) type is always pre-initialized. I think our implemention will always do that.

Reply:
It is `asdf` itself (the return value of `uninitializedDefaultValue`) that is null. The NPE is thrown when trying to create an instance of a primitive class from `null`
the traceback from OpenJDK is:
```
Exception in thread "main" java.lang.NullPointerException: Cannot cast to null-free type "QOtherPoint;" because the return value of "jdk.internal.misc.Unsafe.uninitializedDefaultValue(java.lang.Class)" is null
	at Point.main(Point.java:10)
```
and the traceback from my code is:
```
Exception in thread "main" java.lang.NullPointerException
	at Point.main(Point.java:10)
	at java.lang.invoke.LambdaForm$DMH/0x00000000a201c020.invokeStatic(LambdaForm$DMH)
	at java.lang.invoke.LambdaForm$MH/0x00000000a202a220.invoke(LambdaForm$MH)
	at java.lang.invoke.LambdaForm$MH/0x00000000a202b220.invokeExact_MT(LambdaForm$MH)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invokeImpl(DirectMethodHandleAccessor.java:155)
```

I also tried to run `uninitializedDefaultValue` on increasingly larger value types, but the NPE did not go away. Those tests used the following code:

```
import jdk.internal.misc.Unsafe;
import java.lang.reflect.Field;



primitive class Point {
	public static void main(String[] args) throws Throwable {
		Unsafe unsafe = Unsafe.getUnsafe();

		OtherPoint31 asdf = unsafe.uninitializedDefaultValue(OtherPoint31.class);
		System.out.println(asdf.x);
	}
}


primitive class ValueTypeInt {
	ValueTypeInt(int i) {this.i = i;}
	final int i;
}

primitive class OtherPoint { final ValueTypeInt x, y; OtherPoint(ValueTypeInt x, ValueTypeInt y) { this.x = x; this.y = y; } }
primitive class OtherPoint2 { final OtherPoint x, y; OtherPoint2(OtherPoint x, OtherPoint y) { this.x = x; this.y = y; } }
primitive class OtherPoint3 { final OtherPoint2 x, y; OtherPoint3(OtherPoint2 x, OtherPoint2 y) { this.x = x; this.y = y; } }
primitive class OtherPoint4 { final OtherPoint3 x, y; OtherPoint4(OtherPoint3 x, OtherPoint3 y) { this.x = x; this.y = y; } }
primitive class OtherPoint5 { final OtherPoint4 x, y; OtherPoint5(OtherPoint4 x, OtherPoint4 y) { this.x = x; this.y = y; } }
primitive class OtherPoint6 { final OtherPoint5 x, y; OtherPoint6(OtherPoint5 x, OtherPoint5 y) { this.x = x; this.y = y; } }
primitive class OtherPoint7 { final OtherPoint6 x, y; OtherPoint7(OtherPoint6 x, OtherPoint6 y) { this.x = x; this.y = y; } }
primitive class OtherPoint8 { final OtherPoint7 x, y; OtherPoint8(OtherPoint7 x, OtherPoint7 y) { this.x = x; this.y = y; } }
primitive class OtherPoint9 { final OtherPoint8 x, y; OtherPoint9(OtherPoint8 x, OtherPoint8 y) { this.x = x; this.y = y; } }
primitive class OtherPoint10 { final OtherPoint9 x, y; OtherPoint10(OtherPoint9 x, OtherPoint9 y) { this.x = x; this.y = y; } }
primitive class OtherPoint11 { final OtherPoint10 x, y; OtherPoint11(OtherPoint10 x, OtherPoint10 y) { this.x = x; this.y = y; } }
primitive class OtherPoint12 { final OtherPoint11 x, y; OtherPoint12(OtherPoint11 x, OtherPoint11 y) { this.x = x; this.y = y; } }
primitive class OtherPoint13 { final OtherPoint12 x, y; OtherPoint13(OtherPoint12 x, OtherPoint12 y) { this.x = x; this.y = y; } }
primitive class OtherPoint14 { final OtherPoint13 x, y; OtherPoint14(OtherPoint13 x, OtherPoint13 y) { this.x = x; this.y = y; } }
primitive class OtherPoint15 { final OtherPoint14 x, y; OtherPoint15(OtherPoint14 x, OtherPoint14 y) { this.x = x; this.y = y; } }
primitive class OtherPoint16 { final OtherPoint15 x, y; OtherPoint16(OtherPoint15 x, OtherPoint15 y) { this.x = x; this.y = y; } }
primitive class OtherPoint17 { final OtherPoint16 x, y; OtherPoint17(OtherPoint16 x, OtherPoint16 y) { this.x = x; this.y = y; } }
primitive class OtherPoint18 { final OtherPoint17 x, y; OtherPoint18(OtherPoint17 x, OtherPoint17 y) { this.x = x; this.y = y; } }
primitive class OtherPoint19 { final OtherPoint18 x, y; OtherPoint19(OtherPoint18 x, OtherPoint18 y) { this.x = x; this.y = y; } }
primitive class OtherPoint20 { final OtherPoint19 x, y; OtherPoint20(OtherPoint19 x, OtherPoint19 y) { this.x = x; this.y = y; } }
primitive class OtherPoint21 { final OtherPoint20 x, y; OtherPoint21(OtherPoint20 x, OtherPoint20 y) { this.x = x; this.y = y; } }
primitive class OtherPoint22 { final OtherPoint21 x, y; OtherPoint22(OtherPoint21 x, OtherPoint21 y) { this.x = x; this.y = y; } }
primitive class OtherPoint23 { final OtherPoint22 x, y; OtherPoint23(OtherPoint22 x, OtherPoint22 y) { this.x = x; this.y = y; } }
primitive class OtherPoint24 { final OtherPoint23 x, y; OtherPoint24(OtherPoint23 x, OtherPoint23 y) { this.x = x; this.y = y; } }
primitive class OtherPoint25 { final OtherPoint24 x, y; OtherPoint25(OtherPoint24 x, OtherPoint24 y) { this.x = x; this.y = y; } }
primitive class OtherPoint26 { final OtherPoint25 x, y; OtherPoint26(OtherPoint25 x, OtherPoint25 y) { this.x = x; this.y = y; } }
primitive class OtherPoint27 { final OtherPoint26 x, y; OtherPoint27(OtherPoint26 x, OtherPoint26 y) { this.x = x; this.y = y; } }
primitive class OtherPoint28 { final OtherPoint27 x, y; OtherPoint28(OtherPoint27 x, OtherPoint27 y) { this.x = x; this.y = y; } }
primitive class OtherPoint29 { final OtherPoint28 x, y; OtherPoint29(OtherPoint28 x, OtherPoint28 y) { this.x = x; this.y = y; } }
primitive class OtherPoint30 { final OtherPoint29 x, y; OtherPoint30(OtherPoint29 x, OtherPoint29 y) { this.x = x; this.y = y; } }
primitive class OtherPoint31 { final OtherPoint30 x, y; OtherPoint31(OtherPoint30 x, OtherPoint30 y) { this.x = x; this.y = y; } }
```

Reply:
okay can you try this:

```
import jdk.internal.misc.Unsafe;
import java.lang.reflect.Field;

public class Test {


        public static void main(String[] args) throws Throwable {
                System.out.println("start test....");
                Unsafe unsafe = Unsafe.getUnsafe();

                checkNull(unsafe.uninitializedDefaultValue(SmallVT.class), SmallVT.class);
                checkNull(unsafe.uninitializedDefaultValue(LargeVT.class), LargeVT.class);

                new SmallVT(0L);
                new LargeVT(new VT1K(new Long512(0l, 0l, 0l, 0l, 0l, 0l, 0l, 0l), new Long512(0l, 0l, 0l, 0l, 0l, 0l, 0l, 0l)), new VT1K(new Long512(0l, 0l, 0l, 0l, 0l, 0l, 0l, 0l), new Long512(0l, 0l, 0l, 0l, 0l, 0l, 0l, 0l)));

                checkNull(unsafe.uninitializedDefaultValue(SmallVT.class), SmallVT.class);
                checkNull(unsafe.uninitializedDefaultValue(LargeVT.class), LargeVT.class);

                SmallVT smallVT = unsafe.uninitializedDefaultValue(SmallVT.class);
                long v = smallVT.v;

                LargeVT largeVT = unsafe.uninitializedDefaultValue(LargeVT.class);
                long v2 = largeVT.v1.v1.v1;

        }



        static void checkNull(Object o, Class<?> clazz) {
                if (o == null) {
                        System.out.println(clazz.getName() + " is null");
                } else {
                        System.out.println(clazz.getName() + " is not null");
                }
        }

        static primitive class Long512 {
                long v1,v2,v3,v4,v5,v6,v7,v8;

                Long512(long v1, long v2, long v3, long v4, long v5, long v6, long v7, long v8) {
                        this.v1 = v1;
                        this.v2 = v2;
                        this.v3 = v3;
                        this.v4 = v4;
                        this.v5 = v5;
                        this.v6 = v6;
                        this.v7 = v7;
                        this.v8 = v8;
                }
        }

        static primitive class SmallVT {
                long v;

                SmallVT(long v) {
                        this.v = v;
                }
        }

        static primitive class VT1K {
                Long512 v1, v2;

                VT1K(Long512 v1, Long512 v2) {
                        this.v1 = v1;
                        this.v2 = v2;
                }
        }

        static primitive class LargeVT {
                VT1K v1, v2;

                LargeVT(VT1K v1, VT1K v2) {
                        this.v1 = v1;
                        this.v2 = v2;
                }
        }
}
```
result should be 

```
start test....
Test$SmallVT is null
Test$LargeVT is null
Test$SmallVT is not null
Test$LargeVT is not null
```

Reply:
yep, I got the expected result:
```
start test....
Test$SmallVT is null
Test$LargeVT is null
Test$SmallVT is not null
Test$LargeVT is not null
```

Reply:
thanks for verfying